### ISSUE TITLE
imap tests leave reactor in an unclean state

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2864]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#2864|
|Type|defect|
|Created|2007-11-03 22:33:43Z|

```
===============================================================================
[ERROR]: twisted.mail.test.test_imap.Timeout.testLongFetchDoesntTimeout

Traceback (most recent call last):
Failure: twisted.trial.util.DirtyReactorAggregateError: Reactor was unclean.
DelayedCalls: (set twisted.internet.base.DelayedCall.debug = True to debug)
<DelayedCall 143165836 [1799.99969482s] called=0 cancelled=0 SimpleServer.__timedOut()>
-------------------------------------------------------------------------------
```

<details><summary>Searchable metadata</summary>

```
trac-id__2864 2864
type__defect defect
reporter__radix radix
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1194129223000000 1194129223000000
changetime__1194197643000000 1194197643000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix commented|
|-|-|

Ready for review in unclean-imap-2864.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @radix|
|-|-|

testLongFetchDoesntTimeout could be renamed test_longFetchDoesntTimeout and would benefit from a docstring. There are trailing whitespaces in test_imap.

Overall, these tests in ```Timeout``` are pretty bad, and should all use a deterministic clock, but out of scope etc.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_3"></a>|@radix set owner to @therve|
|-|-|

I've fixed the whitespace, fixed test naming convention, removed tearDown (because it was totally unnecessary), and added a large comment to the implementation fix describing the change.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_4"></a>|@radix commented|
|-|-|

Filed a new ticket for using a fake clock: [#2865](https://github.com/twisted/twisted/issues/2865)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @radix|
|-|-|

There is a typo in test_longFetchDoesntTimeout docstring: "connecection"

You can merge after that correction, but test_serverTimeout and test_idleClientDoesDisconnect would love to have some cute docstrings.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_6"></a>|@radix set status to `closed`|
|-|-|

(In [21574]) Merge unclean-imap-2864

Author: jml, radix
Reviewer: therve
Fixes [#2864](https://github.com/twisted/twisted/issues/2864)

test_longFetchDoesntTimeout (n?\195?\169e testLongFetchDoesntTimeout) no longer leaves
the reactor in an unclean state. Some docstrings were added to clarify the
tests and implementation comments were added.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
